instructor/compiler.py

```diff
@@ -11,6 +11,23 @@ def preview_as_json(assignment_id):
 
 
 
+def get_protocol_headers(assignment):
+    headers = ['Strain', 'Treatment']
+    # Optional headers
+    optional_vars = ['Concentration', 'Start Time',
+                     'Duration', 'Temperature',
+                     'Collection Time']
+    # assignment fields names
+    var_fields = ['has_concentration', 'has_start_time',
+                  'has_duration', 'has_temperature',
+                  'has_collection_time']
+
+    # Adding headers for enabled experimental variables
+    for index, field in enumerate(var_fields):
+        if getattr(assignment, field):
+            headers.append(optional_vars[index])
```

Comment:
It would be cleaner to combine `optional_vars` and `var_fields` to be a list of tuples, for example `[('Concentration', 'has_concentration'), ('Start Time', 'has_start_time')`... This would allow you to iterate through it like

    for header, field in pairs:
        ...